importlib_resources/_py3.py

```diff
@@ -94,22 +95,26 @@ def path(
     """
     reader = _common.get_resource_reader(_common.get_package(package))
     return (
-        _path_from_reader(reader, resource)
+        _path_from_reader(reader, _common.normalize_path(resource))
         if reader else
         _common.as_file(
             _common.files(package).joinpath(_common.normalize_path(resource)))
         )
 
 
-@contextmanager
 def _path_from_reader(reader, resource):
-    norm_resource = _common.normalize_path(resource)
+    return _path_from_resource_path(reader, resource) or \
+        _path_from_open_resource(reader, resource)
+
+
+def _path_from_resource_path(reader, resource):
     with suppress(FileNotFoundError):
-        yield Path(reader.resource_path(norm_resource))
-        return
-    opener_reader = reader.open_resource(norm_resource)
-    with _common._tempfile(opener_reader.read, suffix=norm_resource) as res:
-        yield res
+        return Path(reader.resource_path(resource))
+
+
+def _path_from_open_resource(reader, resource):
+    saved = io.BytesIO(reader.open_resource(resource).read())
```

Comment:
Should we be explicitly closing the result of `reader.open_resource()`? (In a with block, presumably)

Reply:
Good suggestion, but on investigation, I see that [open_resource](https://docs.python.org/3/library/importlib.html#importlib.abc.ResourceReader.open_resource) is expected to return a file-like object and I don't see anything in the [spec for file object](https://docs.python.org/3/glossary.html#term-file-object) that stipulates that it must supply a context manager or close method, and since I can't be certain what other providers may be implementing this method and because the prior behavior was to simply call the method without a `close()`, I'm going to retain that behavior for now.